activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -1163,8 +1187,9 @@ def create_table_definition(name, temporary = false, options = nil, as = nil) #
       def binary_to_sql(limit) # :nodoc:
         case limit
         when 0..0xfff;           "varbinary(#{limit})"
-        when nil;                "blob"
-        when 0x1000..0xffffffff; "blob(#{limit})"
+        when nil, 0x1000..0xffff;   'blob'
+        when 0x10000..0xffffff;     'mediumblob'
+        when 0x1000000..0xffffffff; 'longblob'
```

Comment:
This is the same way as [text_to_sql](https://github.com/kamipo/rails/blob/add_text_and_blob_shorthand_methods/activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb#L1209-L1217):

```ruby
      def text_to_sql(limit) # :nodoc:
        case limit
        when 0..0xff;               'tinytext'
        when nil, 0x100..0xffff;    'text'
        when 0x10000..0xffffff;     'mediumtext'
        when 0x1000000..0xffffffff; 'longtext'
        else raise(ActiveRecordError, "No text type has byte length #{limit}")
        end
      end
```

`blob(2147483647)` is not usual. It is `longblob` usually.